code/modules/mob/death.dm

```diff
@@ -96,6 +96,14 @@
 	if(SSticker.mode)
 		SSticker.mode.check_win()
 
+	//This might seems like an useless computation to the programmer of the future, why would we do this?
+	//Easy! That's because otherwise, the hostile AI will keep us referenced, leading to an harddel
+	for(var/mob/living/simple_animal/hostile/hostile_in_sight in get_targets_in_LOS(world.view))
+		hostile_in_sight.targets.Remove(src)
+
+		if(hostile_in_sight.target_mob == src)
+			hostile_in_sight.target_mob = null
```

Comment:
I think this is going to cause a decent amount of lag when mobs are dying. It's not too bad, but I think a preferable solution is using signals here.

Reply:
![image](https://github.com/Aurorastation/Aurora.3/assets/65877598/b5ffe301-de90-4c4f-8afd-9ef5607a2bea)
25.86 picoseconds on average, per mob death, plus however long to set the two variables and perform one check

Using a signal would be optimal for other reasons, the problem is that we do not have (or i was unable to find) a signal that triggers on mob death, tg has COMSIG_LIVING_DEATH that we're wholly missing